paywall/src/__tests__/unlock.js/Wallet/setupUserAccounts.test.ts

```diff
@@ -26,36 +24,62 @@ describe('Wallet.setupUserAccounts()', () => {
     },
     unlockUserAccounts: true,
   }
-  const startup: StartupConstants = {
-    network: 1984,
-    debug: 0,
-    paywallUrl: 'http://paywall',
-    accountsUrl: 'http://app/accounts',
-    erc20ContractAddress: '0x591AD9066603f5499d12fF4bC207e2f577448c46',
-  }
 
-  function makeWallet() {
+  let setUserAccountAddress = jest.fn()
+  let setUserAccountNetwork = jest.fn()
+  let createIframeSpy: any
+
+  beforeAll(() => {
+    fakeWindow = new FakeWindow()
     iframes = new IframeHandler(
       fakeWindow,
       dataIframeUrl,
       checkoutIframeUrl,
       userIframeUrl
     )
-    iframes.init(config)
-    wallet = new Wallet(fakeWindow, iframes, config, startup)
-    wallet.setupUserAccounts()
-  }
+    createIframeSpy = jest.spyOn(iframes.accounts, 'createIframe')
+    setupUserAccounts({
+      iframes,
+      config,
+      setUserAccountAddress,
+      setUserAccountNetwork,
+    })
+  })
 
-  function testingWallet() {
-    return wallet as any
-  }
+  it('should create the accounts iframe', () => {
+    expect.assertions(1)
 
-  beforeEach(() => {
-    fakeWindow = new FakeWindow()
-    makeWallet()
+    expect(createIframeSpy).toHaveBeenCalled()
+  })
+
+  it('should handle READY from the accounts iframe', () => {
+    expect.assertions(2)
+
+    fakeWindow.receivePostMessageFromIframe(
```

Comment:
It's pretty cumbersome to work with these test fixtures. Would be much nicer to be able to test at a higher level of abstraction, and moving to a post message library will allow us to do that.